### PR TITLE
GrafanaData: Fix week start for non-English browsers

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
This PR fixes the week start behavior on browsers using another language than English. The issue was that `weekdays()` return the week days in the set locale and we compare it with values from the backend that are always in English (at least for now); so we couldn't find the week day index and setting the week start in the preferences or the dashboard settings didn't work. 

**Special notes for your reviewer**:
I didn't find a better way to ensure that the `getWeekdayIndex` function doesn't change the locale but I'm not very familiar with `moment`. I couldn't find anything else in the doc though.

### CODE REVIEW

**Review by spinillos — APPROVED** [review]

File: packages/grafana-data/src/datetime/moment_wrapper.ts [hunk 1]
```diff
@@ -123,7 +123,12 @@ export const dateTimeForTimeZone = (
 };
 
 export const getWeekdayIndex = (day: string) => {
-  return moment.weekdays().findIndex((wd) => wd.toLowerCase() === day.toLowerCase());
+  const prevLocale = getLocale();
+  setLocale('en');
+  const index = moment.weekdays().findIndex((wd) => wd.toLowerCase() === day.toLowerCase());
+  setLocale(prevLocale);
+
+  return index;
```
Comment (leventebalogh) [inline packages/grafana-data/src/datetime/moment_wrapper.ts:131]: **Suggestion:** although I am usually against using comments, I think here we could leave some explanation on the _why_ (as that's probably going to get "lost" later).

Alternatively - as the server returns these in English (currently) - we could also just do a much dumber, although simpler option:
```javascript
export const getWeekdayIndexByEnglishName(day: string) => [
    'Sunday', 
    'Monday', 
    'Tuesday', 
    'Wednesday', 
    'Thursday', 
    'Friday', 
    'Saturday'
].findIndex(wd => wd.toLowerCase() === day.toLowerCase());
```

Comment (AgnesToulet) [reply packages/grafana-data/src/datetime/moment_wrapper.ts]: Great suggestion, thanks! It's easier to understand and there are less unintended side-effect risks. I made the update.

**Review by leventebalogh — COMMENTED** [review]
LGTM 👍 

(I am not sure that `getWeekdayIndex()` is still used anywhere else?)

**Review by leventebalogh — APPROVED** [review]

**Comment by leventebalogh** [comment]
You are absolutely right @AgnesToulet, sorry, I somehow missed that. 👍

**Pull request merged**